src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -2130,6 +2263,116 @@ public void run() {
         return proceedToValidation;
     }
 
+    /**
+     * Returns true if the current Problem is configured to use a sandbox, sandbox usage is supported on the current platform, 
+     * and the current client is not a Team; false otherwise.
+     * If the returned value is True, the sandbox for the problem will have been copied into the Execute directory.
+     * 
+     *TODO: this is a query method with a side-effect (copying the sandbox into the execute directory); that's not good style...
+     *TODO:  Need to refactor this into two steps:  a query (without side effects) asking whether the problem is supposed to use
+     *TODO:    a sandbox, and then if that returns True, a call to a separate method which installs the sandbox.  This is tricky
+     *TODO:    because we want to get the proper exception back to the caller...
```

Comment:
As noted above, I think the refactoring indicated by the above comments should be done as part of this PR.

Reply:
Fixed in [this commit](https://github.com/pc2ccs/pc2v9/pull/732/commits/376601787954cfe6acce71cc171ee64b1e9241b0)